mountpoint-s3-fs/src/superblock/inode.rs

```diff
@@ -240,6 +244,7 @@ pub struct InodeState {
     pub stat: InodeStat,
     pub write_status: WriteStatus,
     pub kind_data: InodeKindData,
+    pub completion_hook: Option<CompletionHook>,
```

Comment:
Let's document what this hook does, when it is created and when it is triggered.